scipy/stats/_hypotests.py

```diff
@@ -377,15 +376,22 @@ def cramervonmises(rvs, cdf, args=()):
 
 def _get_wilcoxon_distr(n):
     """
-    Distribution of counts of the Wilcoxon ranksum statistic r_plus (sum of
+    Distribution of probability of the Wilcoxon ranksum statistic r_plus (sum of
     ranks of positive differences).
-    Returns an array with the counts/frequencies of all the possible ranks
+    Returns an array with the probabilities of all the possible ranks
     r = 0, ..., n*(n+1)/2
     """
-    cnt = _wilcoxon_data.COUNTS.get(n)
-
-    if cnt is None:
-        raise ValueError("The exact distribution of the Wilcoxon test "
-                         "statistic is not implemented for n={}".format(n))
-
-    return np.array(cnt, dtype=int)
+    result = []
+    c = np.ones(1, dtype=np.double)
+    result.append(c)
+
+    # uint64max = 2**64 - 1
+    for k in range(1, n + 1):
```

Comment:
Is there a formal reference for this algorithm, or just based on @WarrenWeckesser s comments?

Reply:
The implementation is based on @WarrenWeckesser s comments, looking at R's source code and the following paper: https://www.researchgate.net/publication/286887362_Normal_Approximations_to_the_Distributions_of_the_Wilcoxon_Statistics_Accurate_to_What_N_Graphical_Insights